src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -361,37 +361,42 @@ def get_raw_token(self, resource=None, scopes=None, subscription=None, tenant=No
             raise CLIError("Please specify only one of subscription and tenant, not both")
 
         account = self.get_subscription(subscription)
-        resource = resource or self.cli_ctx.cloud.endpoints.active_directory_resource_id
 
         identity_type, identity_id = Profile._try_parse_msi_account_name(account)
         if identity_type:
-            # MSI
+            # managed identity
             if tenant:
-                raise CLIError("Tenant shouldn't be specified for MSI account")
-            msi_creds = MsiAccountTypes.msi_auth_factory(identity_type, identity_id, resource)
-            msi_creds.set_token()
-            token_entry = msi_creds.token
-            creds = (token_entry['token_type'], token_entry['access_token'], token_entry)
+                raise CLIError("Tenant shouldn't be specified for managed identity account")
+            from .auth.util import scopes_to_resource
+            msi_creds = MsiAccountTypes.msi_auth_factory(identity_type, identity_id,
+                                                         scopes_to_resource(scopes))
+            sdk_token = msi_creds.get_token(*scopes)
```

Comment:
The old implementation of ADAL-based Azure CLI returns `token_entry` containing inconsistent fields:

- `expires_on` for managed identity
- `expiresOn` for ADAL credential

They are unified later at **command module** level:

https://github.com/Azure/azure-cli/blob/df737edb429a03a16794f8eb8d63435507c84562/src/azure-cli/azure/cli/command_modules/profile/custom.py#L82-L85

We use `get_token` to unify them to epoch int `expires_on` in **core** instead.

Reply:
Just out of curiosity, will `token_entry['expiresOn']` has its own value in some cases before it is overwritten by `_fromtimestamp(int(token_entry['expires_on'])).strftime("%Y-%m-%d %H:%M:%S.%f")`?
Their values should be the same in all cases, but only in different formats, right?

Reply:
No. `get_token` only returns `AccessToken` which **must** have int epoch `expires_on`.